cadence/contracts/NFTRetrieval.cdc

```diff
@@ -148,6 +148,24 @@ pub contract NFTRetrieval {
         return false
     }
 
+    pub fun getInventory(address:Address) : {String: [UInt64]} {
+        let inventory : {String:[UInt64]}={}
+        let account=getAccount(address)
+        let collections : {String:PublicPath} ={}
+        let types = NFTCatalog.getCatalogTypeData()
+        for nftType in types.keys {
+            let typeData=types[nftType]!
+            let collectionKey=typeData.keys[0]
```

Comment:
All the collections are stored in the same path, we just want the ids of them all. Paritioning these into subcollections will take up a lot of gas since you need to borrow each one.